core/block/editor/template.go

```diff
@@ -56,27 +51,19 @@ func (t *Template) CreationStateMigration(ctx *smartblock.InitContext) migration
 	})
 }
 
-// GetNewPageState returns state that can be safely used to create the new document
-// it has not localDetails set
-func (t *Template) GetNewPageState(name string) (st *state.State, err error) {
-	st = t.NewState().Copy()
-
-	if err = t.updateTypeKey(st); err != nil {
-		return nil, err
-	}
-
-	st.RemoveDetail(bundle.RelationKeyTargetObjectType.String(), bundle.RelationKeyTemplateIsBundled.String())
-	st.SetDetailAndBundledRelation(bundle.RelationKeySourceObject, pbtypes.String(t.Id()))
-	// clean-up local details from the template state
-	st.SetLocalDetails(nil)
-
-	if name != "" {
-		st.SetDetail(bundle.RelationKeyName.String(), pbtypes.String(name))
-		if title := st.Get(template.TitleBlockId); title != nil {
-			title.Model().GetText().Text = ""
+func (t *Template) UpdateTypeKey(st *state.State) error {
+	objectTypeID := pbtypes.GetString(st.Details(), bundle.RelationKeyTargetObjectType.String())
+	if objectTypeID != "" {
+		typeKey, err := t.getTypeKeyById(objectTypeID)
```

Comment:
https://linear.app/anytype/issue/GO-2191/refactor-object-store-add-method-getuniquekeybyid

may be do it now?

Reply:
Done!